events/sink.go

```diff
@@ -56,3 +64,49 @@ func (s *eventSink) Write(evt goevents.Event) error {
 func (s *eventSink) Close() error {
 	return nil
 }
+
+func adaptEvent(topic string, evt Event) filters.Adaptor {
+	return filters.AdapterFunc(func(fieldpath []string) (string, bool) {
+		if len(fieldpath) == 0 {
+			return "", false
+		}
+
+		s := evt.(*sinkEvent)
+
+		switch fieldpath[0] {
```

Comment:
So, to clarify, the adaptor framework allows a type to implement adaptor. I created the AdaptorFunc approach for the lazy or when you just want to test something.

I'm not sure about this approach, and that is why I haven't started implementing adaptor on various types.